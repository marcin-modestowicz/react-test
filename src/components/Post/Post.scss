article {
    border: 1px solid gray;
    padding: 1rem;

    h3 {
        text-align: right;
    }
}