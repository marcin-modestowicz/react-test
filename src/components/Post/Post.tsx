import React from 'react';

interface Props {
    post: {
        title: string;
        author: string;
        content: string;
    }
}

const Post = ({ post: { title, author, content }}): Props => (
    <article>
        <header>
            <h2>{title}</h2>
        </header>
    </article>
);

export default Post;
