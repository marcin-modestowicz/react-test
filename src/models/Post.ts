export interface ApiData {
    userId: number;
    id: number;
    title: string;
    body: string;
}

class Post {
    userId: number;
    id: number;
    title: string;
    body: string;

    constructor(apiData: ApiData) {
        const {
            userId,
            id,
            title,
            body,
        } = apiData;

        this.id = id;
        this.userId = userId;
        this.title = title;
        this.body = body;
    }
}

export default Post;
