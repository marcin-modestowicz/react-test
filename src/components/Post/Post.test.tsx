import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Post from "./Post";

describe("Post component", () => {
  test("should render", () => {
    const post = shallow(
        <Post
            title="title"
            author="author"
            content="content"
        />
    );
    expect(toJson(post)).toMatchSnapshot();
  });
});
