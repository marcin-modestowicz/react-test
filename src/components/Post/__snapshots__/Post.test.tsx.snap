// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component should render 1`] = `
<article>
  <header>
    <h2>
      title
    </h2>
    <h3>
      author
    </h3>
  </header>
  content
</article>
`;
